File: tests/csharp_sample_project/ItemProcessor.cs
// tests/sample_project2/ItemProcessor.cs

namespace SampleProject2;

/// <summary>
/// Processes individual Item objects based on configured rules.
/// </summary>
public class ItemProcessor
{
    private readonly int _threshold;

    /// <summary>
    /// Initializes the ItemProcessor with a processing threshold.
    /// </summary>
    public ItemProcessor(int threshold)
    {
        _threshold = threshold;
        Console.WriteLine($"ItemProcessor initialized with threshold: {_threshold}");
    }

    /// <summary>
    /// Processes a single item, marking it as processed and applying logic.
    /// </summary>
    /// <param name="item">The Item object to process.</param>
    /// <returns>True if processing was successful.</returns>
    public bool ProcessItem(Item item)
    {
        Console.WriteLine($"Processing item ID: {item.ItemId}, Name: '{item.Name}', Value: {item.Value}");

        if (item.Value > _threshold)
        {
            Console.WriteLine($"Item '{item.Name}' (ID: {item.ItemId}) value {item.Value:F2} exceeds threshold {_threshold}.");
        }
        else
        {
            Console.WriteLine($"Item '{item.Name}' (ID: {item.ItemId}) value {item.Value:F2} is within threshold {_threshold}.");
        }

        item.MarkAsProcessed();
        return true;
    }
}

File: tests/ruby_sample_project/item_processor.rb
# tests/sample_project2/item_processor.rb

require_relative 'item'

#
# Processes individual Item objects based on configured rules.
#
class ItemProcessor
  # Initializes the ItemProcessor with a processing threshold.
  # @param threshold [Integer] The numerical threshold for processing logic.
  def initialize(threshold)
    @threshold = threshold
    puts "ItemProcessor initialized with threshold: #{@threshold}"
  end

  # Processes a single item, marking it as processed and applying logic.
  # @param item [Item] The Item object to process.
  # @return [Boolean] True if processing was successful.
  def process_item(item)
    puts "Processing item ID: #{item.item_id}, Name: '#{item.name}', Value: #{item.value}"

    if item.value > @threshold
      puts "Item '#{item.name}' (ID: #{item.item_id}) value #{item.value} exceeds threshold #{@threshold}."
    else
      puts "Item '#{item.name}' (ID: #{item.item_id}) value #{item.value} is within threshold #{@threshold}."
    end

    item.mark_as_processed
    true
  end
end